python/ml4ir/model/losses/pointwise_losses.py

```diff
@@ -10,18 +11,24 @@ def _make_loss_fn(self, **kwargs):
         Define a sigmoid cross entropy loss
         Additionally can pass in record positions to handle positional bias
 
-        NOTE:
-        Should handle different types of scoring functions.
-        Keeping it simple for now.
         """
-        cce = losses.CategoricalCrossentropy()
+        bce = losses.BinaryCrossentropy(reduction=Reduction.SUM_OVER_BATCH_SIZE)
+        mask = kwargs.get("mask")
 
         def _loss_fn(y_true, y_pred):
             # Mask the predictions to ignore padded records
-            mask = kwargs.get("mask")
-            y_true = math.multiply(tf.cast(mask, tf.float32), tf.cast(y_true, tf.float32))
-            y_pred = math.multiply(tf.cast(mask, tf.float32), tf.cast(y_pred, tf.float32))
+            y_true = tf.gather_nd(y_true, tf.where(tf.equal(mask, tf.constant(1.0))))
+            y_pred = tf.gather_nd(y_pred, tf.where(tf.equal(mask, tf.constant(1.0))))
 
-            return cce(y_true, y_pred)
+            return bce(y_true, y_pred)
 
         return _loss_fn
+
+    def _final_activation_op(self):
+        # Pointwise sigmoid loss
+        sigmoid = layers.Activation("sigmoid", name="ranking_scores")
+
+        def sigmoid_op(logits, mask):
+            return sigmoid(logits)
```

Comment:
could the body of this method just be:

`return lambda logits, mask: layers.Activation("sigmoid", name="ranking_scores")(logits)`

Reply:
I mean... yes. But this is more readable, right?